sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -138,12 +139,24 @@ def _is_readonly(p):
 class AzureJSONEncoder(JSONEncoder):
     """A JSON encoder that's capable of serializing datetime objects and bytes."""
 
+    def __init__(self, *args, exclude_readonly: bool = True, exclude_none: bool = False, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
these should both default to `False`. In our generated code, we can just pass `body.to_dict(exclude_readonly=True, exclude_none=True)` to the service

Reply:
I think for generated code, we should use `exclude_none=False` to follow the current logic.

Reply:
talked this through with @lmazuel and @johanste and the reason we want to default it to `True` is we don't want it to be lossy by default and we want to store all information

Reply:
But set `exclude_none` to `True` will drop all null properties, which should be lossy, right?

Reply:
yeah we currently drop all `None` properties when we send to the service, not `azure.core.serialization.NULL`